app/cdap/components/WranglerGrid/NestedMenu/MenuItemComponent/index.tsx

```diff
@@ -40,29 +40,47 @@ export interface IMenuItem {
   icon?: OverridableComponent<SvgIconTypeMap<{}, 'svg'>>;
   toolName?: string;
   open?: boolean;
+  getUsage?: ({ selectedColumnName, selectedFunction }: IGetUsageParams) => string;
+}
+
+export interface IGetUsageParams {
+  selectedColumnName: string | boolean;
+  selectedFunction: IMenuItem;
 }
 
 export interface IMenuItemComponentProps {
   item: IMenuItem;
   index: number;
-  onMenuClick: (e: React.MouseEvent<Element, MouseEvent>, item: IMenuItem) => void;
+  onMenuOptionClick: (e: React.MouseEvent<Element, MouseEvent>, item: IMenuItem) => void;
   columnType: string;
+  hasSimilarType?: boolean;
+  allColumnsType: string;
 }
 
-export default function({ item, index, onMenuClick, columnType }: IMenuItemComponentProps) {
+export default function({
+  item,
+  index,
+  onMenuOptionClick,
+  columnType,
+  hasSimilarType,
+  allColumnsType,
+}: IMenuItemComponentProps) {
   const includesCheck = !(
     item?.supportedDataType?.includes(columnType) || item?.supportedDataType?.includes('all')
   );
   const filteredDataOptionCheck = DATATYPE_OPTIONS.filter(
     (el) =>
-      (el.value === item.value && item.value === columnType.toLowerCase()) ||
-      (item.value === 'integer' && columnType.toLowerCase() === 'int')
+      (el.value === item.value && item.value === columnType?.toLowerCase()) ||
+      (item.value === 'integer' && columnType?.toLowerCase() === 'int')
   ).length;
 
   const getMenuItemDisablProp = () => {
     if (columnType) {
       return includesCheck || filteredDataOptionCheck;
     }
+    if (allColumnsType.toLowerCase() === item.label) {
```

Comment:
integer may fail here, plz check if any fix is required or not

Reply:
this usecase is covered